### DISCUSSION TITLE
Custom script doesn't work when run as service

### DISCUSSION THREAD

**Discussion by TaylorFinklea** [FelixKratz/SketchyBar#523] [Q&A]
I'm stumped and hoping to figure out what I'm doing wrong. I have a custom weather script that I've written, it works great when I run `sketchybar` as my own user. If I use a service either through nix or using `brew services start sketchybar`, the script does not work. Everything is self contained in the script, it does API calls but all secrets are saved in the script, so it should not be relying on env vars. Any ideas of what might be wrong? I've attached screenshots.

My config
![image](https://github.com/FelixKratz/SketchyBar/assets/6810234/fcbffdcd-eb38-43e7-be18-6915ae499c95)

plugins/weather.sh
![SCR-20240320-luyu](https://github.com/FelixKratz/SketchyBar/assets/6810234/4aa2fe17-4f97-4977-9f8b-93885be969f5)

items/weather.sh
![image](https://github.com/FelixKratz/SketchyBar/assets/6810234/9153cc95-06e6-49cd-91c7-aa777e4edddf)

Running the script directly
![SCR-20240320-ltao](https://github.com/FelixKratz/SketchyBar/assets/6810234/b896c752-e41c-4760-bb78-fc758a5b51a6)

Running as a service
![SCR-20240320-lsxj](https://github.com/FelixKratz/SketchyBar/assets/6810234/c756aa07-869c-4fc6-82fd-f8900943b247)
![SCR-20240320-lsvc](https://github.com/FelixKratz/SketchyBar/assets/6810234/430b35a6-47d9-480e-8f1f-aa9e765e1918)

**Comment by FelixKratz** [comment]
Sounds like a PATH issue. The PATH available to sketchybar when run as a service is severely limited. Try using absolute paths to binaries you might have installed somewhere „uncommon“.

You can check the logs of the sketchybar service in /opt/homebrew/var/log/sketchybar which should reveal the exact reason why the script fails.

**Reply by thoth2357** [reply]
if you find yourself trying to get the logs of sketchybar when being executed as a service and you're using a non ARM apple device like intel. The logs would be located at /usr/local/var/log/sketchybar.
